### PR TITLE
feat: upgrade to rules_js 2.0

### PR DESCRIPTION
**Pull request by gregmagolan**
---

### Type of change

- New feature or functionality (change which adds functionality)

**For changes visible to end-users**

- Breaking change (this change will force users to change their own code or config)
- Suggested release notes are provided below:

### Test plan

- Covered by existing test cases

### CODE REVIEW

File: swc/private/swc.bzl [hunk 1]
```diff
@@ -42,7 +42,14 @@ _attrs = {
         If out_dir is also specified, it is used as the name of the output directory.
         Otherwise, the directory is named the same as the target.""",
     ),
-    "data": js_lib_helpers.JS_LIBRARY_DATA_ATTR,
```
Comment (jbedard) [inline swc/private/swc.bzl:45]: Did we delete this?

Comment (gregmagolan) [reply swc/private/swc.bzl:45]: Yes. I opted out of the pattern of exporting individual attributes and their docstrings. Its cleaner and easier to just reference back to the upstream attribute in the downstream attribute docstring.

File: swc/defs.bzl [hunk 2]
```diff
@@ -146,9 +146,14 @@ def swc_plugin(name, src = None, config = {}, **kwargs):
     if not types.is_dict(config):
         fail("config must be a dict, not a " + type(config))
 
+    # For backward compat
```
Comment (jbedard) [inline swc/defs.bzl:149]: Won't rules_js 2.0 be a breaking change and we don't need this? Should we print a warning?

File: swc/dependencies.bzl [hunk 3]
```diff
@@ -15,20 +15,27 @@ def http_archive(**kwargs):
...
         name = "aspect_rules_js",
-        sha256 = "7b2a4d1d264e105eae49a27e2e78065b23e2e45724df2251eacdd317e95bfdfd",
-        strip_prefix = "rules_js-1.31.0",
-        url = "https://github.com/aspect-build/rules_js/releases/download/v1.31.0/rules_js-v1.31.0.tar.gz",
+        sha256 = "b627acf34f928507dfabae69d3fd8702ce3d45bed568ad4c44b243cb789f08c6",
+        strip_prefix = "rules_js-2.0.0-alpha.2",
+        url = "https://github.com/aspect-build/rules_js/releases/download/v2.0.0-alpha.2/rules_js-v2.0.0-alpha.2.tar.gz",
+    )
+
+    http_archive(
```
Comment (jbedard) [inline swc/dependencies.bzl:36]: Why is this added?

File: swc/defs.bzl (hunk 2 above)
Comment (jbedard) [reply swc/defs.bzl:149]: Warning and then we can drop it on the next major? 👍 

I do think we should minimize breaking changes in rulesets when we are forcing so many updates all at once. The rules_js breaking changes are enough already.

**Review by alexeagle — APPROVED** [review]
